WalletWasabi.Fluent/ViewModels/AddWallet/AddedWalletPageViewModel.cs

```diff
@@ -50,9 +50,20 @@ protected override void OnNavigatedTo(bool isInHistory, CompositeDisposable disp
 	{
 		base.OnNavigatedTo(isInHistory, disposables);
 
-		if (!Services.WalletManager.WalletExists(_keyManager.MasterFingerprint))
+		if (ShouldAddWallet())
 		{
 			Services.WalletManager.AddWallet(_keyManager);
 		}
 	}
+
+	private bool ShouldAddWallet()
+	{
+		var isWatchOnly = _keyManager.MasterFingerprint is null;
+		if (isWatchOnly)
+		{
+			return false;
+		}
```

Comment:
`KeyManager` has already an `IsWatchOnly` property.

Reply:
Great! I'm using it already :)